protocol/network/7-properties.md

```diff
@@ -1,16 +1,105 @@
+7 Properties
+============
 
-# IPFS Protocol on the Wire
+## 7.1 Communication Model - Streams
 
-Authors: [Juan Benet](http://github.com/jbenet)
+The Network layer handles all the problems of connecting to a peer, and exposes
+simple bidirectional streams. Users can both open a new stream
+(`NewStream()`) and register a stream handler (`SetStreamHandler`). The user
+is then free to implement whatever wire messaging protocol she desires. This
+makes it easy to build peer-to-peer protocols, as the complexities of
+connectivity, multi-transport support, flow control, and so on, are handled.
```

Comment:
as an aside, I will probably be implementing an interface in Go that looks like this:

```go
n := libp2p.NewNode(...all the config...)
proto, err := n.Mount("/dht") // or n.Bind
stream1, err := proto.Dial()
stream2, err := proto.Dial()
stream3, err := proto.Dial()
stream5, err := proto.Accept() // error because not listening yet
err := proto.Listen()
stream5, err := proto.Accept()
stream6, err := proto.Accept()
stream7, err := proto.Accept()
```

Aaaaand reroll (another interface)

```
// use Mount or Bind with Dial and Listen modes
n := libp2p.NewNode(...all the config...)
proto1, err := n.Bind("/echo", Listen)
proto2, err := n.Bind("/bitswap", DialAndListen)
proto3, err := n.Bind("/ping", Dial)
stream1, err := proto1.Dial() // error because proto1 is only bound to Listen
stream2, err := proto1.Dial() // ok
stream3, err := proto2.Dial() // ok
stream4, err := proto2.Accept() // ok
stream5, err := proto3.Dial() // ok
stream6, err := proto3.Accept() // error because proto2 is only bound to Dial
```
- bind is more bsd socket-y
- mount is more unix fs-y
- mount is more what _I_ like and mean.
- mount evokes filesystem endpoint pathspace, which is what we're exposing here
- mount sounds more plan9 friendly.
- i'll probably do mount unless someone can point out to me why Bind and Mount should be different steps.

cc @whyrusleeping for thoughts.